packages/cheqd-sdk/src/ledger/CheqdSdkLedgerService.ts

```diff
@@ -0,0 +1,107 @@
+import type { AbstractCheqdSDKModule, CheqdSDK, ICheqdSDKOptions, DidStdFee, DIDDocument } from '@cheqd/sdk'
+import type { SignInfo } from '@cheqd/ts-proto/cheqd/did/v2'
+import type { MsgCreateResourcePayload } from '@cheqd/ts-proto/cheqd/resource/v2'
+
+import { injectable } from '@aries-framework/core'
+import { createCheqdSDK, DIDModule, ResourceModule, CheqdNetwork } from '@cheqd/sdk'
+import { DirectSecp256k1HdWallet, DirectSecp256k1Wallet } from '@cosmjs/proto-signing'
+import { EnglishMnemonic as _ } from '@cosmjs/crypto'
+import { fromString } from 'uint8arrays'
+
+import { CheqdSdkModuleConfig } from '../CheqdSdkModuleConfig'
+
+export interface CheqdSdkLedgerConfig {
+  networkType?: CheqdNetwork
+  rpcUrl?: string
+  cosmosPayerSeed?: string
+}
+
+export enum DefaultRPCUrl {
+  Mainnet = 'https://rpc.cheqd.net',
+  Testnet = 'https://rpc.cheqd.network',
+}
+
+export class EnglishMnemonic extends _ {
+  public static readonly _mnemonicMatcher = /^[a-z]+( [a-z]+)*$/
+}
+
+@injectable()
+export class CheqdSdkLedgerService {
+  private readonly cosmosPayerWallet: Promise<DirectSecp256k1HdWallet | DirectSecp256k1Wallet>
+  private sdk?: CheqdSDK
+  private fee?: DidStdFee
+  private rpcUrl: string | undefined
+
+  public constructor(config: CheqdSdkModuleConfig) {
+    const { cosmosPayerSeed, rpcUrl } = config
+    if (!cosmosPayerSeed || cosmosPayerSeed === '') {
+      this.cosmosPayerWallet = DirectSecp256k1HdWallet.generate()
+      return
+    }
+    this.cosmosPayerWallet = EnglishMnemonic._mnemonicMatcher.test(cosmosPayerSeed)
+      ? DirectSecp256k1HdWallet.fromMnemonic(cosmosPayerSeed, { prefix: 'cheqd' })
+      : DirectSecp256k1Wallet.fromKey(fromString(cosmosPayerSeed.replace(/^0x/, ''), 'hex'), 'cheqd')
+
+    this.rpcUrl = rpcUrl
+  }
+
+  public async connect(options: { network?: string; fee?: DidStdFee } = {}) {
+    const sdkOptions: ICheqdSDKOptions = {
+      modules: [DIDModule as unknown as AbstractCheqdSDKModule, ResourceModule as unknown as AbstractCheqdSDKModule],
+      rpcUrl: this.rpcUrl
+        ? this.rpcUrl
+        : options.network === CheqdNetwork.Mainnet
+        ? DefaultRPCUrl.Mainnet
+        : DefaultRPCUrl.Testnet,
+      wallet: await this.cosmosPayerWallet.catch(() => {
+        throw new Error(`[did-provider-cheqd]: valid cosmosPayerSeed is required`)
+      }),
+    }
+
+    this.sdk = await createCheqdSDK(sdkOptions)
+    this.fee = options.fee
+  }
```

Comment:
Generally the approach we've taken with the indy pool service is that we create pool implemetnation that handle the connection to a specific network. This way you can connect to multiple ledgers simultaneously. E.g. the resolver now calls connect, but in the meantime another resolve action could also call connect and we'd be opening the gate for race conditions. 

Do you think we coudl add that layer where we'd be manaing multiple networks? In the indy implementation we then get the netwrok based on the indyNamespace, which in cheqd's case would be `mainnet` vs `testnet`.